js/script.js

```diff
@@ -0,0 +1,176 @@
+		allBoxes[x].className += " o-cell";
+		isPlayerOne = true;
+		turns++;
+	}
+}
+
+var checkForWinner = function(){   // the following functions check to see if anyone has won
+	if (isWinner('x')){
+		document.getElementById('winner-name').innerHTML = "The Winner is X!";
+		gameOver = true;
+	} else if (isWinner('o')){
+		document.getElementById('winner-name').innerHTML = "The Winner is O!";
+		gameOver = true;
+	} else if (turns >= 9){
+		document.getElementById('winner-name').innerHTML = "NOBODY WINS!";
+		gameOver = true;
+	}
+	
+}
+
+var isWinner = function(player){
+	return winsRow(player) || winsCol(player) || winsDia(player);
+}
+
+var winsRow = function(player){
+	return connectThree(player, allBoxes[0], allBoxes[1], allBoxes[2]) ||
+		   connectThree(player, allBoxes[3], allBoxes[4], allBoxes[5]) ||
+		   connectThree(player, allBoxes[6], allBoxes[7], allBoxes[8]);
+}
+
+var winsCol = function(player){
+	return connectThree(player, allBoxes[0], allBoxes[3], allBoxes[6]) ||
+		   connectThree(player, allBoxes[1], allBoxes[4], allBoxes[7]) ||
+		   connectThree(player, allBoxes[2], allBoxes[5], allBoxes[8]);
+}
+
+var winsDia = function(player){
+	return connectThree(player, allBoxes[0], allBoxes[4], allBoxes[8]) ||
+		   connectThree(player, allBoxes[2], allBoxes[4], allBoxes[6]);
+}
+
+var connectThree = function(player, cell1, cell2, cell3){
+	if(cell1.className.indexOf(player) >= 0 && 
+	   cell2.className.indexOf(player) >= 0 &&
+	   cell3.className.indexOf(player) >= 0) {
+		return player;
+	}
+}
+
+
+var turnCheck = function(){   // this function changes the position of the turn indicator 
+	if(isPlayerOne){
+		document.getElementById("x-span").className = 'active-turn';
+		document.getElementById("o-span").className = '';
+	} else {
+		document.getElementById("x-span").className = '';
+		document.getElementById("o-span").className = 'active-turn'
+	}
+}
+
+var changeBox = function(cell) {     // this assigns a square to x or o
+	if (!gameOver && cell.className.indexOf("cell") === -1) {  // this line prevents overwriting played squares
+		if (isPlayerOne){
+			cell.className += " x-cell";  //gives the square P1 color
+			//cell.innerHTML = "<p>X<p>";   //gives the square an X
+			isPlayerOne = false;          //makes it P2's turn
+			checkForWinner();
+			turns++;
+			computerMove();				//prompts computer to take its turn (sigle-player only!)
+		} else {
+			cell.className += " o-cell";
+			//cell.innerHTML = "<p>O<p>";
+			isPlayerOne = true;	
+			turns++;
+		}
+	}
+	
+	turnCheck();
+}
+
+
+var clearBoard = function(){  //function for the clear button
+	for(var i in allBoxes){
+		allBoxes[i].className = 'play-square'; //clears all x && o-cell classes from boxes
+		allBoxes[i].innerHTML = ''; //clears all text from boxes
+		isPlayerOne = true;         //makes it player one's turn
+		turnCheck();
+		turns = 0;
+		gameOver = false;
+		document.getElementById('winner-name').innerHTML = "Who will survive?";
+	}
```

Comment:
This will usually work, but keep in mind that `for..in` loops are reserved for objects, and could result in weird behaviors when used on arrays. 

See MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#Array_iteration_and_for...in